sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ApplicationGateway.java

```diff
@@ -552,6 +576,48 @@ interface WithHttp2 {
             WithCreate withoutHttp2();
         }
 
+        /**
+         * The stage of the application gateway definition allowing to associate a Web Application Firewall Policy.
+         */
+        interface WithWebApplicationFirewallPolicy {
+            /**
+             * Associates an existing Web Application Firewall Policy with the application gateway.
+             * The new association will remove and replace previous WAF policy association (if any).
+             *
+             * @param wafPolicy existing Web Application Firewall Policy
+             * @return the next stage of the definition
+             */
+            WithCreate withExistingWebApplicationFirewallPolicy(WebApplicationFirewallPolicy wafPolicy);
+
+            /**
+             * Associates an existing Web Application Firewall Policy with the application gateway.
+             * The new association will remove and replace previous WAF policy association (if any).
+             *
+             * @param resourceId existing Web Application Firewall Policy
+             * @return the next stage of the definition
+             */
+            WithCreate withExistingWebApplicationFirewallPolicy(String resourceId);
+
+            /**
+             * Associates a new Web Application Firewall Policy with the application gateway.
+             * The new association will remove and replace previous WAF policy association (if any).
+             *
+             * @param enable whether to enable the policy or not
+             * @param mode the Web Application Firewall mode
+             * @return the next stage of the update
+             */
+            WithCreate withNewWebApplicationFirewallPolicy(boolean enable, WebApplicationFirewallMode mode);
+
+            /**
+             * Associates a new Web Application Firewall Policy with the application gateway.
+             * The new association will remove and replace previous WAF policy association (if any).
+             *
+             * @param creatable full definition of the new Web Application Firewall Policy
+             * @return the next stage of the definition
+             */
+            WithCreate withNewWebApplicationFirewallPolicy(Creatable<WebApplicationFirewallPolicy> creatable);
```

Comment:
I assume the former is a simplified version of latter.

Do we need the `enable` there? Would default enable=true be enough for former?

Reply:
It's to match legacy waf configuration method signature. Default enable=true should be fine.